### PR TITLE
Update Python to 3.8 for Windows

### LINKED ISSUE #59865: [BUG] The Windows package of salt 3002.5 contains multiple third-party software vulnerabilities.
**Description**
The Windows package of salt 3002.5 contains zlib 1.2.2,openssl 1.1.1d,expat 2.2.7,curl 7.68.0,sqlite3 3.28.0, contains vulnerabilities.

**Setup**
https://repo.saltstack.com/windows/Salt-Minion-3002.5-Py3-x86-Setup.exe

**Expected behavior**
Upgrading Components to Clear Vulnerabilities

**Versions Report**
salt-3002.5

```
PASTE HERE
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Uses Python 3.8 as the python environment for Salt on Windows
Updates the libsodium.dll to version 1.0.18
Updates the OpenSSL dlls to version 1.1.1k
Enables pre-commit for Python 3.8 for the static dependencies
Fixes the build.bat and build_pkg.bat to NOT overwrite existing environment variables if they have been created by CI/CD
No longer patches the easy_install binaries as they have been deprecated with the latest setuptools
Adds 3.8 to the clean_env.bat

https://github.com/saltstack/salt-jenkins/pull/1698

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Issues
https://github.com/saltstack/salt/issues/59865

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Ch3LL — DISMISSED** [review]
Looks good. The tests are still running python 3.7. I'm assuming the golden images will need to be updated outside of this salt project. Is that correct?

File: changelog/59865.fixed [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Update the Salt installer to use Python 3.8 for its python environment
+Update libsodium.dll to 1.0.18
+Update OpenSSL dlls to 1.1.1k
```
Comment (waynew) [inline changelog/59865.fixed:3]: **question**: Does` towncrier handle this sort of entry properly? If not, I'd suggest something like:

> Update Windows build deps & DLLs, Use Python 3.8, libsodium.dll 1.0.18, OpenSSL dlls to 1.1.1k

(that should be `1.1.1k`?)

File: setup.py [hunk 2]
```diff
@@ -501,19 +501,21 @@ def indent_log():
...
+                "openssl/1.1.1k/ssleay32.dll",
+                "openssl/1.1.1k/libeay32.dll",
+                "libsodium/1.0.18/libsodium.dll",
+            ):
                 # See if the library is already on the system
                 if find_library(fname):
                     continue
                 furl = url.format(bits=platform_bits[:2], fname=fname)
-                fdest = dest.format(fname=fname)
+                fdest = dest.format(fname=fname.split("/")[-1])
```
Comment (waynew) [inline setup.py:516]: **suggestion (non-blocking)**: would `os.path.basename(fname)` be better here?

**Review by waynew — DISMISSED** [review]
just a couple of nits, otherwise it looks great!

File: setup.py [hunk 3]
```diff
@@ -501,19 +501,21 @@ def indent_log():
                 yield
 
         platform_bits, _ = platform.architecture()
-        url = "https://repo.saltstack.com/windows/dependencies/{bits}/{fname}.dll"
-        dest = os.path.join(os.path.dirname(sys.executable), "{fname}.dll")
+        url = "https://repo.saltstack.com/windows/dependencies/{bits}/{fname}"
```
Comment (s0undt3ch) [inline setup.py:504]:
```suggestion
        url = "https://repo.saltproject.io/windows/dependencies/{bits}/{fname}"
```
@bryceml right?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (twangboy) [reply setup.py]: Oui

**Review by s0undt3ch — DISMISSED** [review]

**Review by krionbsd — DISMISSED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Review by krionbsd — DISMISSED** [review]

**Comment by Ch3LL** [comment]
re-run full all

**Comment by twangboy** [comment]
Looks like we're hitting the Windows path length limit of 260: https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation


```
>>> test = "C:\\Users\\Administrator\\AppData\\Local\\Temp\\kitchen\\testing\\.nox\\pytest-parametrized-3-crypto-none-transport-zeromq-coverage-true\\Lib\\site-packages\\jedi\\third_party\\typeshed\\third_party\\2and3\\requests\\packages\\urllib3\\packages\\ssl_match_hostname\\_implementation.pyi"
>>> len(test)
263
```

File: tests/conftest.py [hunk 4]
```diff
@@ -117,6 +118,20 @@ class LiveLoggingStreamHandler(
 
 
 # ----- PyTest Tempdir Plugin Hooks --------------------------------------------------------------------------------->
+def pytest_tempdir_temproot():
```
Comment (s0undt3ch) [inline tests/conftest.py:121]: This change is no longer required 

Latest version of salt-factories has this issue addressed.

https://github.com/saltstack/pytest-salt-factories/blob/0.906.0/src/saltfactories/plugins/__init__.py#L19

**Review by s0undt3ch — DISMISSED** [review]

File: tests/conftest.py (hunk 4 above)
Comment (twangboy) [reply tests/conftest.py:121]: So, I can just delete this then?

Comment (s0undt3ch) [reply tests/conftest.py]: Yes, please

**Review by Ch3LL — DISMISSED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Comment by Ch3LL** [comment]
re-run full all

**Comment by Ch3LL** [comment]
ping @twangboy looks like the same test failed on both windows OSs when running a full test run.

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**